stackslib/src/net/p2p.rs

```diff
@@ -5224,6 +5302,116 @@ impl PeerNetwork {
         Ok(())
     }
 
+    /// Load up the parent stacks tip.
+    /// For epoch 2.x, this is the pointer to the parent block of the current stacks tip
+    /// For epoch 3.x, this is the pointer to the tenure-start block of the parent tenure of the
+    /// current stacks tip.
+    /// If this is the first tenure in epoch 3.x, then this is the pointer to the epoch 2.x block
+    /// that it builds atop.
+    pub(crate) fn get_parent_stacks_tip(
```

Comment:
`get_header_by_coinbase_height` takes a `&mut StacksDBTx` (a flavor of `IndexDBTx`), which requires opening a transaction against the chainstate.  This can't be avoided without either substantial refactoring or code duplication, since that `&mut StacksDBTx` is used to perform a MARF read.  This code is used by block-processing, so it needs to support taking a `&mut StacksDBTx`.  But, there's currently no way to convert an `IndexDBTx` into an `IndexDBConn`, which means that the p2p thread would need to open a transaction against the chainstate (which itself is a no-no because it could either stall block processing, or cause the p2p thread to get stalled by block processing).  I'd rather leave the code in place.